deployment/lib/cf-url-rewriter.js

```diff
@@ -0,0 +1,59 @@
+const aws = require('aws-sdk');
+
+const s3 = new aws.S3({ apiVersion: '2006-03-01' });
+
+exports.handler = async (event, context) => {
+
+    const request = event.Records[0].cf.request;
+    request.uri = request.uri.replace(/^\/ci\/...\//, '\/')
+    callback(null, request);
+
+    // Below is the working in progress logic to get the max build number under a version.
+    // Hardcode the version to be 1.2.0 and bucket name to test-access-1-20 for demo purpose.
+    console.log('Received event:', JSON.stringify(event, null, 2));
+
+    // Get the object from the event and show its content type
+    const bucket = "test-access-1-20";
+
+    // Create the parameters for calling listObjects
+    var bucketParams = {
+        Bucket: bucket,
+        Prefix: '1.2.0/',
+        Delimiter: '/'
+    };
+
+    try {
```

Comment:
Seems like we could author unit test for this more complex logic to using jest

Reply:
Please don't over engineering here

Reply:
Writing a unit test is certainly not over-engineering. +1 to Peter's ask.

Reply:
UT is ok. However I'd like to see incremental change and working version to make progress on automation to unlock manual release testing.

I would also like to see actual code than pseudo code or comments

Talking is cheap

Reply:
Unit tests added in latest code after the functional strategy was finalized and implemented. Thanks for all for the input! Really appreciated it!